src/OhadSoft.AzureLetsEncrypt.Renewal/Util/CertificateHelper.cs

```diff
@@ -15,6 +15,9 @@ namespace OhadSoft.AzureLetsEncrypt.Renewal.Util
 {
     internal static class CertificateHelper
     {
+        internal static string[] LetsEncryptIssuerNames { get; set; } = new[] { "Let's Encrypt Authority X1", "Let's Encrypt Authority X2", "Let's Encrypt Authority X3", "Let's Encrypt Authority X4", "R3", "R4", "E1", "E2" };
```

Comment:
I meant that these should be `new HashSet<string>(StringComparer.Ordinal)`

Reply:
and you might as well make it a private `readonly` field, no need for a property not a setter

Reply:
The way I understand it (please correct me if I'm wrong), @sjkp's letsencrypt-siteextension is the source for the package LetsEncrypt.Azure.Core (not the V2 one) from which you use the [RenewCertificate](https://github.com/sjkp/letsencrypt-siteextension/blob/6b9f2e1db98efb8897a2a76d417065c604d91354/LetsEncrypt.SiteExtension.Core/CertificateManager.cs#L141) method. 
Therein @sjkp does the same certificate issuer check against the old list of names which will make the whole thing fail again.

I got it to work on my side by using my own fork of that project with the new issuer check. 

Unless @sjkp updates the library or this project handles the certificate request completely by itself I think this whole thing won't work.

As for the new HashSet<string>, I think for such a small collection of string there is no advantage in using a HashSet over a bare array. I've changed the issuer names to be an private static IEnumerable<string> backed by an array now. You are right, there is no real reason why these would need to be changed from outside.

Reply:
You are right about everything except one thing - there is a simple way to handle `renewXNumberOfDaysBeforeExpiration` while still letting `letsencrypt-siteextension` do the heavy lifting. I pushed the required code to this PR , check it out and if you get a few minutes please test it!

Reply:
Regarding the `HashSet`
1. You're probably right about the perf diff being small, but because we're doing the comparison in a tight loop for all the certificates in the subscription, with no idea how many there could be, I prefer to be safe than sorry
2. I'm already using `HashSet` a few lines below for the thumbprints so I prefer to stay consistent
3. Semantically/declaratively a set is the correct data structure, as we don't expect repetitions and it is used for quick lookup

None of the above are probably super critical but I was editing the code anyway so I went ahead and changed it :)

Reply:
> 
> 
> You are right about everything except one thing - there is a simple way to handle `renewXNumberOfDaysBeforeExpiration` while still letting `letsencrypt-siteextension` do the heavy lifting. I pushed the required code to this PR , check it out and if you get a few minutes please test it!

Makes sense, well done!